thruster_control/src/thruster_control.cpp

```diff
@@ -108,8 +123,67 @@ ThrusterControl::ThrusterControl(ros::NodeHandle& nodeHandle)
   canIntf.SetEnableCANLogging(currentLoggingEnabled);
   canIntf.AddCanNodeIdToList(canNodeId1);
   canIntf.AddCanNodeIdToList(canNodeId2);
+
   publisher_reportBatteryHealth = diagnostic_tools::create_publisher<sensor_msgs::BatteryState>(
       nodeHandle, "/thruster_control/report_battery_health", 1);
+
+  diagnostic_tools::PeriodicMessageStatusParams paramsReportsBatteryHealthCheckPeriod{};
+  paramsReportsBatteryHealthCheckPeriod.min_acceptable_period(minReportBatteryHealthRate);
+  paramsReportsBatteryHealthCheckPeriod.max_acceptable_period(maxReportBatteryHealthRate);
+  diagnostic_tools::Diagnostic diagnosticBatteryHealthInfoStale(diagnostic_tools::Diagnostic::WARN,
+                                                                ReportFault::BATTERY_INFO_STALE);
+  paramsReportsBatteryHealthCheckPeriod.abnormal_diagnostic(diagnosticBatteryHealthInfoStale);
+  diagnosticsUpdater.add(
+      publisher_reportBatteryHealth.add_check<diagnostic_tools::PeriodicMessageStatus>(
+          "rate check", paramsReportsBatteryHealthCheckPeriod));
+
+  diagnostic_tools::MessageStagnationCheckParams paramsBatteryMessageStagnationcheck;
+  diagnostic_tools::Diagnostic diagnosticBatteryInfoStagnate(
+      diagnostic_tools::Diagnostic::WARN, ReportFault::BATTERY_INFO_STAGNATED);
+  paramsBatteryMessageStagnationcheck.stagnation_diagnostic(diagnosticBatteryInfoStagnate);
+
+  diagnosticsUpdater.add(
+      publisher_reportBatteryHealth.add_check<diagnostic_tools::MessageStagnationCheck>(
+          "stagnation check",
+          [batteryCurrentSteadyBand](const sensor_msgs::BatteryState& a,
+                                     const sensor_msgs::BatteryState& b)
+          {
+            return std::fabs(a.current - b.current < batteryCurrentSteadyBand);
```

Comment:
@patapufate I presume you meant
```suggestion
            return std::fabs(a.current - b.current) < batteryCurrentSteadyBand;
```
?

Reply:
yes.